src/qibolab/result.py

```diff
@@ -18,7 +18,8 @@ def __init__(self, data: np.ndarray):
         self.voltage: npt.NDArray[np.complex128] = data
 
     def __add__(self, data):
-        return self.__class__(np.append(self.voltage, data.voltage))
+        """Concatenates two results for sweeping variables."""
+        return self.__class__(np.column_stack([self.voltage, data.voltage]))
```

Comment:
I know that at least the qblox driver relies on the current behavior of `__add__`, and this change will cause it to produce data with different shape than it does now. The use case for the qblox driver is to append two 1D arrays to make one big 1D array.

Reply:
Another use case in the qblox driver, where two results are combined is to append two 2D arrays along axis 0. For this the `__add__` method is not used and appending is done manually for now as a workaround. See https://github.com/qiboteam/qibolab/pull/831 for more details.

Reply:
So in a nutshell, different parties want to combine results in at least three different ways. One simple `__add__` method cannot rule them all. I think the only correct resolution is to remove the implementation of the `__add__` method, and instead implement a proper 'append' method, or some classmethod that accepts two instances and returns a new instance with combined results. And this method/classmethod would accept additional arguments to determine how exactly to do the combining of the two pieces. I actually attempted to do this as part of the above linked PR, but ended up not doing it because things are more complicated than they appear. This is mostly because of the existence of so many classes to represent results, which makes it hard to figure out how to cleanly implement appending functionality. I think the result classes should be revisited and simplified/properly designed first.

Reply:
I see, so there was indeed some other drivers that used this. I would say that, in general, it's not ideal that this part of the result object is not covered y tests at all.

For the moment at least, I will implement the np.columnstack directly in the RFSoC driver

Reply:
> For the moment at least, I will implement the np.columnstack directly in the RFSoC driver

That's definitely the best, and I'd recommend even for the current Qblox.

> This is mostly because of the existence of so many classes to represent results

And this is also an issue on its own, since it complicates the results handling.

I'd suggest going back to a scheme with a single kind of result, and that being a NumPy array of suitable dimension (the meaning should be clear to the caller). Subclassing/wrapping NumPy array once would not be a problem, but if not needed, I would avoid even that.
During the transition, I'd recommend handling results in the drivers as pure NumPy arrays until returning them, and do the conversion (wrapping) at the very end.

Reply:
@hay-k, I removed the changes to the results so this should be ok